<?xml version="1.0"?>
<launch>

  <group> 

	  <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
	  <arg name="robot_env" default="$(optenv ROBOT_ENV !!NO_ROBOT_ENV_SET!!)"/>
	  <arg name="map" default="$(find cob_default_env_config)/$(arg robot_env)/map.yaml" />

	  <!--- Run tf publihser -->
	  <node name="gt_tf_pub" pkg="nav_ground_truth_publisher" type="tf_pub.py" args="$(arg map)" output="screen">
      <param name="~pose_topic" value="gt_amcl_pose"/>
      <param name="~base_frame" value="gazebo_gt"/>
      <param name="~global_frame" value="map"/>
	  </node>

	  <!--- Run map_server -->
	  <node name="gt_map_server" pkg="map_server" type="map_server" args="$(arg map)" output="screen">
		  <remap from="map_metadata" to="gt_map_metadata" />
		  <remap from="map" to="gt_map" />
	  </node>

	  <!--- Run AMCL -->
	  <!-- run scan unifier -->
    <node pkg="cob_scan_unifier" type="scan_unifier_node" name="gt_scan_unifier" output="screen">
      <rosparam file="$(find cob_navigation_config)/$(arg robot)/scan_unifier_config.yaml" command="load"/>
      <remap from="scan_unified" to="/gt_scan_unified"/>
    </node>

	  <!-- start amcl -->
	  <node pkg="amcl" type="amcl" name="gt_amcl" output="screen">
		  <remap from="scan" to="gt_scan_unified" />
		  <remap from="amcl_pose" to="gt_amcl_pose" />
		  <param name="odom_model_type" value="omni"/>
		  <param name="transform_tolerance" value="0.2" />
		  <param name="gui_publish_rate" value="-1.0"/>
		  <param name="laser_max_beams" value="30"/>
		  <param name="laser_max_range" value="29.5"/>
		  <param name="min_particles" value="100"/>
		  <param name="max_particles" value="5000"/>
		  <param name="update_min_d" value="0.05"/>
		  <param name="update_min_a" value="0.5"/>
		  <param name="kld_err" value="0.01"/>
		  <param name="kld_z" value="0.99"/>
		  <param name="odom_alpha1" value="0.2"/>
		  <param name="odom_alpha2" value="0.2"/>
		  <!-- translation std dev, m -->
		  <param name="odom_alpha3" value="0.2"/>
		  <param name="odom_alpha4" value="0.2"/>
		  <param name="odom_alpha5" value="0.2"/>
		  <param name="laser_z_hit" value="0.95"/>
		  <param name="laser_z_short" value="0.1"/>
		  <param name="laser_z_max" value="0.05"/>
		  <param name="laser_z_rand" value="0.05"/>
		  <param name="laser_sigma_hit" value="0.2"/>
		  <param name="laser_lambda_short" value="0.1"/>
		  <param name="laser_model_type" value="likelihood_field"/>
		  <param name="laser_likelihood_max_dist" value="2.0"/>
		  <param name="odom_frame_id" value="/odom_combined"/>
		  <param name="base_frame_id" value="/base_footprint"/>
		  <param name="global_frame_id" value="/map"/>
		  <param name="resample_interval" value="2"/>
		  <param name="transform_tolerance" value="0.1"/>
		  <param name="recovery_alpha_slow" value="0.0"/>
		  <param name="recovery_alpha_fast" value="0.0"/>
		  
		  <!-- switching of the tf broadcast ( we are publishing it by hand ) -->
		  <param name="tf_broadcast" value="false"/>
	  </node>
	  
	</group>  
	
</launch>
